app/src/main/java/au/id/neasbey/biblecast/util/BibleAPIResponseParser.java

```diff
@@ -0,0 +1,165 @@
+package au.id.neasbey.biblecast.util;
+
+import android.text.Html;
+import android.text.Spanned;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.List;
+
+/**
+ * Created by craigneasbey on 30/06/15.
+ *
+ * Bible API response parser utility
+ */
+public class BibleAPIResponseParser {
+
+    private static final String responseKey = "response";
+    private static final String searchKey = "search";
+    private static final String resultKey = "result";
+    private static final String typeKey = "type";
+    private static final String passagesKey = "passages";
+    private static final String versesKey = "verses";
+    private static final String referenceKey = "reference";
+    private static final String textKey = "text";
+
+    /**
+     * Parse the JSON response from bibles.org
+     * @param jsonString JSON response in format: http://bibles.org/pages/api/documentation/search
+     * @param spannedList List of HTML elements
+     * @throws Exception Allows JSON parse exceptions to be translated for the app
+     */
+    public void parseJSONToList(String jsonString, List<Spanned> spannedList) throws Exception {
+
+        try {
+            boolean results = false;
+
+            // Creates a new JSONObject with name/value mappings from the JSON string
+            JSONObject jsonValues = new JSONObject(jsonString);
+
+            // Get response values
+            JSONObject responseValues = jsonValues.optJSONObject(responseKey);
+
+            // Get search values
+            if(responseValues != null) {
+                JSONObject searchValues = responseValues.optJSONObject(searchKey);
+
+                // Get result values
+                if(searchValues != null) {
+                    JSONObject resultValues = searchValues.optJSONObject(resultKey);
+
+                    // Get type value
+                    if(resultValues != null) {
+                        String typeValue = resultValues.optString(typeKey);
+
+                        // check type
+                        switch (typeValue) {
+                            case passagesKey:
+                                // Get passages values
+                                results = parsePassages(spannedList, resultValues.optJSONArray(passagesKey));
+                                break;
+                            case versesKey:
+                                // Get verses values
+                                results = parseVerses(spannedList, resultValues.optJSONArray(versesKey));
+                                break;
+                            default:
+                                results = parseVerses(spannedList, resultValues.optJSONArray(versesKey));
+                        }
+                    }
+                }
+            }
+
+            if(!results) {
+                throw new Exception("No results found");
+            }
+        } catch (JSONException e) {
```

Comment:
Again, I prefer reworked exception handling.
* Throwing JSONException = bad as it ties your code/API to the JSON library
* Throwing Exception = bas as it forces code to catch Exception
* Throwing your own Exception is good, as you know it came from your own code and the JSONException can still be wrapped as the cause.